src/command/pin.rs

```diff
@@ -1,14 +1,22 @@
+use failure::Fail;
 use serde::Deserialize;
 
 use notion_core::session::{ActivityKind, Session};
 use notion_core::style::{display_error, ErrorContext};
 use notion_core::tool::ToolSpec;
 use notion_core::version::VersionSpec;
-use notion_fail::{ExitCode, Fallible};
+use notion_fail::{throw, ExitCode, Fallible, NotionFail};
+use notion_fail_derive::*;
 
 use crate::command::{Command, CommandName, Help};
 use crate::Notion;
 
+/// Thrown when a user tries to `notion pin` something other than node/yarn/npm.
+#[derive(Debug, Fail, NotionFail)]
+#[fail(display = "Only node, yarn, and npm can be pinned in a project")]
+#[notion_fail(code = "InvalidArguments")]
+pub struct CannotPinPackageError;
```

Comment:
This should probably be converted to the new `ErrorDetails` enum.

Reply:
You're right, I missed that one